lms/services/course.py

```diff
@@ -140,19 +140,33 @@ def search(  # noqa: PLR0913, PLR0917
             h_userid=h_userid,
         ).all()
 
-    def get_organization_courses(
-        self, organization: Organization, h_userid: str | None
+    def get_courses(
+        self,
+        h_userid: str | None,
+        organization: Organization | None = None,
     ):
-        courses_query = self._search_query(
-            organization_ids=[organization.id],
-            h_userid=h_userid,
-            limit=None,
-        )
-        return (
+        """Get a list of unique courses.
+
+        :param organization: organization the courses belong to.
+        :param h_userid: only courses this user has access to.
+        """
+        courses_query = (
+            self._search_query(
+                organization_ids=[organization.id] if organization else None,
+                h_userid=h_userid,
+                limit=None,
+            )
             # Deduplicate courses by authority_provided_id, take the last updated one
-            courses_query.distinct(Course.authority_provided_id)
+            .distinct(Course.authority_provided_id)
             .order_by(Course.authority_provided_id, Course.updated.desc())
-            .all()
+            # Only select the ID of the deduplicated courses
+        ).with_entities(Course.id)
+
+        return (
+            self._db.query(Course)
+            .filter(Course.id.in_(courses_query))
+            # We can sort these again without affecting deduplication
```

Comment:
To check my understanding, the reason for using two queries here rather than doing the search with just one, is to enable using one sort order for the inner query, and then a different sort order for the outer query?